### PR TITLE
DTM-15545

### PR DESCRIPTION
**Pull request by brenthosie**
## Description

- Update dependencies.
- Make build scripts more clear.
- Enforce test & lint during commit.

## Related Issue

https://jira.corp.adobe.com/browse/DTM-15545

## Motivation and Context

Dependabot alerted us that we had dependencies out of date. Additionally, enforced some changes in the build pipeline.

## How Has This Been Tested?

-Added test file for `createGetSyntheticEventMeta.js` (untested module)
- ran `npm test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Dependency updates 

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,14 @@
   "description": "Launch rule engine which processes rules on client websites and delegates logic to extensions.",
   "license": "Apache-2.0",
   "scripts": {
-    "build": "rollup -c --intro \"if (!window.atob) { console.warn('Adobe Launch is unsupported in IE 9 and below.'); return; }\" && uglifyjs dist/engine.js -c -m -o dist/engine.min.js",
-    "build:watch": "rollup --watch -c --intro \"if (!window.atob) { console.warn('Adobe Launch is unsupported in IE 9 and below.'); return; }\" && uglifyjs dist/engine.js -c -m -o dist/engine.min.js",
+    "build": "npm run clean && rollup -c && npm run minify",
+    "build:watch": "npm run clean && rollup --watch -c",
```
Comment (brenthosie) [inline package.json:8]: removed the `minify` step here because in `watch` mode it wasn't working as expected anyways.

File: package.json [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
+    "clean": "rm -rf dist",
+    "minify": "uglifyjs dist/engine.js -c -m -o dist/engine.min.js",
     "prepublishOnly": "npm run build",
-    "test": "karma start && npm run lint",
```
Comment (brenthosie) [inline package.json:10]: lint is now happening in `husky pre-commit`

File: rollup.config.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
-import resolve from 'rollup-plugin-node-resolve';
-import commonjs from 'rollup-plugin-commonjs';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
 
 export default {
   input: 'src/index.js',
   output: {
+    intro: "if (!window.atob) { console.warn('Adobe Launch is unsupported in IE 9 and below.'); return; }",
```
Comment (brenthosie) [inline rollup.config.js:7]: pulled out of `package.json` and placed here.

File: package.json [hunk 4]
```diff
@@ -30,29 +32,38 @@
...
-    "rollup-plugin-node-resolve": "^5.2.0",
+    "rollup": "^2.36.1",
     "uglify-js": "^3.9.4",
-    "webpack": "^4.43.0",
+    "webpack": "^4.46.0",
     "yargs": "^15.3.1"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run lint && npm test"
```
Comment (dompuiu) [inline package.json:63]: On a few other projects I did things a little different. I would run prettier and lint of pre-commit. And run the tests on pre-push: https://github.com/adobe/reactor-turbine-edge/blob/master/package.json#L23-L24.

Comment (dompuiu) [reply package.json:63]: I saw the pattern that I used in other external projects, and I like it. Tests are usually slow to run. The prettier and eslint fix commands are run only on the staged files.

**Review by dompuiu — APPROVED** [review]

**Pull request merged**